File: Products/sample-android/app/src/main/java/mockifer/example/android/edit/EditCatActivity.java
package mockifer.example.android.edit;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.google.gson.Gson;

import java.io.IOException;

import mockifer.example.android.R;
import mockifer.example.android.cats.Cat;
import mockifer.example.android.common.Requester;
import okhttp3.Call;
import okhttp3.Callback;
import okhttp3.Response;

public class EditCatActivity extends AppCompatActivity {
    public static final String EXTRA_CAT = "cat";

    Button deleteButton;
    Button saveButton;
    TextView catIdTextView;
    EditText catNameEditText;
    EditText catAgeEditText;

    Cat cat;

    @Override public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_edit_cat);

        catIdTextView = (TextView) findViewById(R.id.catId);
        catNameEditText = (EditText) findViewById(R.id.catName);
        catAgeEditText = (EditText) findViewById(R.id.catAge);

        cat = (Cat) getIntent().getSerializableExtra(EXTRA_CAT);

        catIdTextView.setText(cat.id);
        catNameEditText.setText(cat.name);
        catAgeEditText.setText(String.valueOf(cat.age));

        deleteButton = (Button) findViewById(R.id.deleteButton);
        saveButton = (Button) findViewById(R.id.saveButton);

        deleteButton.setOnClickListener(new View.OnClickListener() {
            @Override public void onClick(View view) {
                deleteCat();
            }
        });

        saveButton.setOnClickListener(new View.OnClickListener() {
            @Override public void onClick(View view) {
                saveCat();
            }
        });
    }

    void deleteCat() {
        Toast.makeText(this, "Deleting cat...", Toast.LENGTH_SHORT).show();

        Requester.doRequest("DELETE", "/cats/" + cat.id, null, new Callback() {
            @Override public void onFailure(Call call, IOException e) {

            }

            @Override public void onResponse(Call call, final Response response) throws IOException {
                runOnUiThread(new Runnable() {
                    @Override public void run() {
                        if (!response.isSuccessful()) {
                            Toast.makeText(EditCatActivity.this, "Error deleting cat", Toast.LENGTH_SHORT).show();
                        }
                        try { response.body().string(); } catch (IOException e) {}
                        finish();
                    }
                });
            }
        });
    }

    void saveCat() {
        Toast.makeText(this, "Saving cat...", Toast.LENGTH_SHORT).show();

        Cat catToSave = new Cat();
        catToSave.id = cat.id;
        catToSave.name = catNameEditText.getText().toString();
        catToSave.age = Integer.valueOf(catAgeEditText.getText().toString());
        catToSave.image = cat.image;

        Requester.doRequest("PUT", "/cats", new Gson().toJson(catToSave), new Callback() {
            @Override public void onFailure(Call call, IOException e) {

            }

            @Override public void onResponse(Call call, final Response response) throws IOException {
                runOnUiThread(new Runnable() {
                    @Override public void run() {
                        if (!response.isSuccessful()) {
                            Toast.makeText(EditCatActivity.this, "Error updating cat", Toast.LENGTH_SHORT).show();
                        }
                        try {
                            response.body().string();
                        } catch (IOException e) {
                        }
                        finish();
                    }
                });
            }
        });
    }
}

File: support/samples/sample-android/app/src/main/java/io/github/marcelbraghetto/mockifer/example/android/edit/EditCatActivity.kt
package io.github.marcelbraghetto.mockifer.example.android.edit

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.widget.Toast
import com.google.gson.Gson
import io.github.marcelbraghetto.mockifer.example.android.R
import io.github.marcelbraghetto.mockifer.example.android.cats.Cat
import io.github.marcelbraghetto.mockifer.example.android.common.Requester
import kotlinx.android.synthetic.main.activity_edit_cat.*
import okhttp3.Call
import okhttp3.Callback
import okhttp3.Response
import java.io.IOException

class EditCatActivity : AppCompatActivity() {
    private lateinit var cat: Cat

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_edit_cat)

        cat = intent.getSerializableExtra(EXTRA_CAT) as Cat

        catIdTextView.text = cat.id
        catNameEditText.setText(cat.name)
        catAgeEditText.setText(cat.age.toString())

        deleteButton.setOnClickListener { deleteCat() }
        saveButton.setOnClickListener { saveCat() }
    }

    private fun deleteCat() {
        Toast.makeText(this, "Deleting cat...", Toast.LENGTH_SHORT).show()

        Requester.doRequest("DELETE", "/cats/" + cat.id, null, object : Callback {
            override fun onFailure(call: Call, e: IOException) {

            }

            @Throws(IOException::class)
            override fun onResponse(call: Call, response: Response) {
                runOnUiThread {
                    if (!response.isSuccessful) {
                        Toast.makeText(this@EditCatActivity, "Error deleting cat", Toast.LENGTH_SHORT).show()
                    }
                    try {
                        response.body()!!.string()
                    } catch (e: IOException) {
                    }

                    finish()
                }
            }
        })
    }

    private fun saveCat() {
        Toast.makeText(this, "Saving cat...", Toast.LENGTH_SHORT).show()

        val catToSave = Cat(
                id = cat.id,
                name = catNameEditText.text.toString(),
                age = Integer.valueOf(catAgeEditText.text.toString()),
                image = cat.image
        )

        Requester.doRequest("PUT", "/cats", Gson().toJson(catToSave), object : Callback {
            override fun onFailure(call: Call, e: IOException) {

            }

            @Throws(IOException::class)
            override fun onResponse(call: Call, response: Response) {
                runOnUiThread {
                    if (!response.isSuccessful) {
                        Toast.makeText(this@EditCatActivity, "Error updating cat", Toast.LENGTH_SHORT).show()
                    }
                    try {
                        response.body()!!.string()
                    } catch (e: IOException) {
                    }

                    finish()
                }
            }
        })
    }

    companion object {
        const val EXTRA_CAT = "cat"
    }
}
